src/models/partnership/partnership.type.ts

```diff
@@ -0,0 +1,22 @@
+type Partner = {
+    id: string;
+    name: string;
+    logo: string;
+    position: number;
+    link: string;
+};
+
+type Partnership = {
+    id: string;
+    name: string;
+    position: number;
+    partners: Partner[];
+};
+
+type PartnershipDoc = {
+        id: string;
+        name: string;
+        position: number;
+};
```

Comment:
If you need type related to the Firestore Document you can create it inside the data layer. 
In the model directory we want to have only the model used by the presentation layer and agnostic to Firebase types.